test.js

```diff
@@ -1,147 +1,106 @@
-// Simulate an API call to fetch tasks from a server
-function fetchTasks() {
-  return new Promise((resolve) => {
-    setTimeout(() => {
-      resolve([
-        { id: 1, text: 'Finish JavaScript project', completed: false },
-        { id: 2, text: 'Read a book', completed: true },
-        { id: 3, text: 'Go grocery shopping', completed: false },
-      ]);
-    }, 1000);
-  });
-}
-
-// DOM Elements (Missing element IDs, some will not work)
-const taskList = document.getElementById('task-list');
-const taskInput = document.getElementById('taskinput'); // Typo here: 'task-input' is the actual ID
-const addButton = document.getElementById('add-button');
-const errorMessage = document.getElementById('error-message');
-
-// Task storage (in-memory for now)
-let tasks = [];
-
-// Function to render tasks to the screen
-function renderTasks() {
-  taskList.innerHTML = ''; // Clear the list before rendering
-
-  if (tasks.length === 0) {
-    taskList.innerHTML = '<li>No tasks to show</li>';
-  } else {
-    tasks.forEach(task => {
-      const taskElement = document.createElement('li');
-      taskElement.className = task.completed ? 'completed' : '';
-
-      const taskText = document.createElement('span');
-      taskText.textContent = task.text;
-      taskText.className = 'task-text';
-      taskElement.appendChild(taskText);
-
-      // Edit Button
-      const editButton = document.createElement('button');
-      editButton.textContent = 'Edit';
-      editButton.className = 'edit-btn';
-      editButton.onclick = () => editTask(task.id);
-      taskElement.appendChild(editButton);
-
-      // Delete Button (Forgot to add a delete button handler)
-      const deleteButton = document.createElement('button');
-      deleteButton.textContent = 'Delete';
-      deleteButton.className = 'delete-btn';
-      taskElement.appendChild(deleteButton);
-
-      // Mark as Completed Button
-      const completeButton = document.createElement('button');
-      completeButton.textContent = task.completed ? 'Undo' : 'Complete';
-      completeButton.className = 'complete-btn';
-      completeButton.onclick = () => toggleComplete(task.id);
-      taskElement.appendChild(completeButton);
-
-      taskList.appendChild(taskElement);
-    });
-  }
-}
-
-// Function to add a new task
-function addTask(text) {
-  if (!text.trim()) {
-    showError('Task cannot be empty.');
+// A non-optimized to-do list application with redundant logic and poor performance
+let toDoList = [];
+// Function to add a task (overcomplicated and inefficient)
+function addTask(task) {
+  if (task === "") {
+    alert("Task cannot be empty!");
     return;
   }
-
-  const newTask = {
-    id: Date.now(), // Using timestamp as a unique ID
-    text: text,
-    completed: false,
+  let taskId = Date.now(); // Using Date.now() as a unique ID, not optimal for large data
+  let taskObject = {
```

Comment:
Consider using a more efficient method for DOM manipulation, such as batching updates or using a virtual DOM library.